### PR TITLE
Upgrading dependencies

### PR DESCRIPTION
**Pull request by AdrieanKhisbe**
Investigating #202 I realised that yargs was really outdated as long as some other dependencies.
Here is an upgrade solving that. test are unbroken.

Node 8 is also added to the travis build

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ node_js:
   - "4"
   - "6"
   - "7"
+  - "8"
```
Comment (mhamann) [inline .travis.yml]: Given that Node.js 8 isn't *quite* LTS, let's also add it to the `allow_failures` matrix below.

**Comment by mhamann** [comment]
Thanks for the PR! Going to leave this open until we get all of the `next-minor` items merged (only two remaining, soon to be one) and put this into the `next-major` release just in case of unforeseen side-effects produced by such major version jumps.

Would estimate we can merge it within a week or so.

File: .travis.yml (hunk 1 above)
Comment (AdrieanKhisbe) [reply .travis.yml]: done

**Comment by AdrieanKhisbe** [comment]
okey dokey =)

**Comment by mhamann** [comment]
Ready to pull this in, if you could rebase to resolve conflicts, @AdrieanKhisbe.

**Comment by AdrieanKhisbe** [comment]
I'll also add node 9 as accepted failure instead of 8, now that 8 is LTS

**Comment by AdrieanKhisbe** [comment]
done

**Pull request merged**